### PR TITLE
drivers: serial: nrfx_uarte: Deprecate CONFIG_UART_x_NRF_HW_ASYNC

### PR DESCRIPTION
**Pull request by nordic-krch**
Deprecate mode which is using TIMER+(D)PPI for reliable byte counting. Recently a new approach is added (CONFIG_UART_NRFX_UARTE_ENHANCED_RX) which supports reliable byte counting without additional HW resource. This mode is planned to be the only supported RX path mode.

Enhanced RX has slightly different behavior. There are no partial RX packets (events with non-zero offset). There is UART_RX_BUF_RELEASED after each UART_RX_RDY event.

### CODE REVIEW

File: samples/net/cellular_modem/boards/nrf9160dk_nrf9160_ns.conf
Comment (tomi-font) [inline samples/net/cellular_modem/boards/nrf9160dk_nrf9160_ns.conf:1]: Is `UART_NRFX_UARTE_LEGACY_SHIM` (which is a prerequisite of `UART_NRFX_UARTE_ENHANCED_RX`) enabled by default as well?

Comment (nordic-krch) [reply samples/net/cellular_modem/boards/nrf9160dk_nrf9160_ns.conf:1]: yes, legacy shim is the default for all nrf targets. I'm deprecating setting `CONFIG_UART_NRFX_UARTE_LEGACY_SHIM=n` and plan to remove "non-legacy" shim after next Zephyr release.

**Review by tomi-font — DISMISSED** [review]

**Review by anangl — DISMISSED** [review]

**Review by bjarki-andreasen — DISMISSED** [review]

**Review by dcpleung — DISMISSED** [review]

**Comment by kartben** [comment]
@nordic-krch please rebase

**Review by bjarki-andreasen — APPROVED** [review]

**Review by kl-cruz — APPROVED** [review]

**Comment by nordic-krch** [comment]
@dcpleung @anangl can you reapprove. I had to rebase.

**Review by anangl — APPROVED** [review]

**Comment by nordic-krch** [comment]
@dcpleung can you reapprove?

**Comment by nordic-krch** [comment]
@dcpleung can you reapprove?

**Pull request merged**